checker.cpp

```diff
@@ -1,46 +1,144 @@
+        {"Warning: Approaching hypothermia or hyperthermia!", "Warnung: Ann�herung an Unterk�hlung oder Hyperthermie!"},
+        {"Warning: Approaching abnormal pulse rate!", "Warnung: Ann�herung an abnormale Pulsrate!"},
+        {"Warning: Approaching low oxygen saturation!", "Warnung: Ann�herung an niedrige Sauerstoffs�ttigung!"},
+        {"All tests passed.", "Alle Tests bestanden."}
+    }}
+    /* Add more languages here */
+};
+
+void simulateBlinkingEffect() {
+    for (int i = 0; i < 6; i++) {
+        cout << "\r* " << flush;
+        sleep(1);
+        cout << "\r *" << flush;
+        sleep(1);
     }
+}
+
+bool isOutOfRange(float value, const VitalInfo& info) {
+    return value < info.lowerLimit || value > info.upperLimit;
+}
 
-    return 0;
-  } else if(pulseRate < 60 || pulseRate > 100) {
-    cout << "Pulse Rate is out of range!\n";
-    for (int i = 0; i < 6; i++)
-    {
-      cout << "\r* " << flush;
-      sleep(1);
-      cout << "\r *" << flush;
-      sleep(1);
+bool isApproachingWarning(float value, const VitalInfo& info) {
+    float lowerThreshold = info.lowerLimit + (info.upperLimit - info.lowerLimit) * (1 - info.tolerance / 100.0);
+    float upperThreshold = info.upperLimit - (info.upperLimit - info.lowerLimit) * (1 - info.tolerance / 100.0);
+    return value >= lowerThreshold && value <= upperThreshold;
+}
+
+VitalStatus checkVital(VitalType type, float value) {
+    const VitalInfo& info = vitalsInfo[type];
+    if (isOutOfRange(value, info)) {
+        if (isApproachingWarning(value, info)) {
+            cout << translations[currentLanguage][info.warningMessage] << endl;
+            return VitalStatus::Warning;
+        }
+        else {
+            cout << translations[currentLanguage][info.criticalMessage] << endl;
+            return VitalStatus::Critical;
+        }
     }
-    return 0;
-  } else if(spo2 < 90) {
-    cout << "Oxygen Saturation out of range!\n";
-    for (int i = 0; i < 6; i++)
-    {
-      cout << "\r* " << flush;
-      sleep(1);
-      cout << "\r *" << flush;
-      sleep(1);
+    return VitalStatus::Ok;
+}
+
+void setLanguage(Language language) {
+    currentLanguage = language;
+}
+
+string translate(const string& message) {
+    if (translations.find(currentLanguage) != translations.end() &&
+        translations[currentLanguage].find(message) != translations[currentLanguage].end()) {
+        return translations[currentLanguage][message];
     }
-    return 0;
-  }
-  return 1;
+    // If translation not found, return the message in English as a fallback
+    return message;
+}
+
+bool vitalsOk(float temperature, float pulseRate, float spo2) {
+    bool isTemperatureOk = checkVital(VitalType::Temperature, temperature) == VitalStatus::Ok;
+    bool isPulseRateOk = checkVital(VitalType::PulseRate, pulseRate) == VitalStatus::Ok;
+    bool isSpo2Ok = checkVital(VitalType::Spo2, spo2) == VitalStatus::Ok;
```

Comment:
Would this say " Approaching low oxygen saturation" when the SPO2 approaches 100% ?